tool-plugins/vscode/plugin/serverStarter.js

```diff
@@ -106,16 +107,33 @@ function startServices() {
                 onParserError(e);
                 onLSError(e);
             });
-            
+
+            let parserStarted = false;
+            let aggregatedStdout = '';
+            const logLevelDebug = workspace.getConfiguration('ballerina').get('debugLog') === true;
+
             serverProcess.stdout.on('data', (data) => {
                 console.log(`ls: ${data}`);
-                if (`${data}`.indexOf('Parser started successfully') > -1) {
+                if (logLevelDebug) {
+                    outputChannel.append(`${data}`);
+                }
+
+                if (parserStarted) {
+                    return;
+                }
+
+                aggregatedStdout = `${aggregatedStdout}${data}`;
+                if (aggregatedStdout.indexOf('Parser started successfully') > -1) {
                     outputChannel.append('Parser started successfully on port: ' + parserPort + '\n');
                     onParserStarted({port: parserPort});
+                    parserStarted = true;
                 }
             });
             serverProcess.stderr.on('data', (data) => {
                 console.log(`ls: ${data}`);
+                if (logLevelDebug) {
```

Comment:
Shall we move the login to separate function so we need only one if to check the log level.

Reply:
Updated